### ISSUE TITLE
subscriptions.charged_through_date Always has null value

### ISSUE THREAD

**Issue by husam-otri** [killbill/killbill#490]
_subscriptions.charged_through_date_ always has **null**  for billing period Monthly or Annual, this is new bug after i upgrade from 0.14.x to 0.16.x

this is ex for plan which i subscribed to:

```
<plans>
...
<plan name="smallQ-monthly">
            <product>smallQ</product>
            <initialPhases>
                <phase type="TRIAL">
                    <duration>
                        <unit>DAYS</unit>
                        <number>30</number>
                    </duration>
                    <fixed>
                        <fixedPrice><!-- empty price implies $0 -->
                        </fixedPrice>
                    </fixed>
                </phase>
            </initialPhases>
            <finalPhase type="EVERGREEN">
                <duration>
                    <unit>UNLIMITED</unit>
                </duration>
                <recurring>
                    <billingPeriod>MONTHLY</billingPeriod>
                    <recurringPrice>
                        <price>
                            <currency>USD</currency>
                            <value>5.00</value>
                        </price>
                    </recurringPrice>
                </recurring>
                <usages>
                    <usage name="smallQ-monthly-usage" billingMode="IN_ARREAR" usageType="CONSUMABLE">
                        <billingPeriod>MONTHLY</billingPeriod>
                        <tiers>
                            <tier>
                                <blocks>
                                    <tieredBlock>
                                        <unit>words-count</unit>
                                        <size>100000</size>
                                        <prices>
                                            <price>
                                                <currency>USD</currency>
                                                <value>0</value>
                                            </price>
                                        </prices>
                                        <max>100000</max>
                                    </tieredBlock>
                                </blocks>
                            </tier>
                            <tier>
                                <blocks>
                                    <tieredBlock>
                                        <unit>words-count</unit>
                                        <size>10000</size>
                                        <prices>
                                            <price>
                                                <currency>USD</currency>
                                                <value>1.00</value>
                                            </price>
                                        </prices>
                                        <max>1000000000</max>
                                    </tieredBlock>
                                </blocks>
                            </tier>
                        </tiers>
                    </usage>
                </usages>
            </finalPhase>
        </plan>
...
</plans>`
```

**Comment by sbrossie** [comment]
@husam-otri I see that your `Plan` has a `TRIAL`. Is the `charged_through_date` null when you are in `TRIAL` phase or is it always null even as you move to `EVERGREEN` phase? Does it have anything to do with usage (or if you remove usage section) you get same behavior?

**Comment by husam-otri** [comment]
I've tested the subscription with all the following plans :

1- Plan with trial and with usage
2- Plan without trial and with usage
3- plan without trial and without usage


*charged_through_date* always null

please take a look on the plans test cases

<plan name="smallQ-monthly">
    <product>smallQ</product>
    <initialPhases>
        <phase type="TRIAL">
            <duration>
                <unit>DAYS</unit>
                <number>30</number>
            </duration>
            <fixed>
                <fixedPrice>
                </fixedPrice>
            </fixed>
        </phase>
    </initialPhases>
    <finalPhase type="EVERGREEN">
        <duration>
            <unit>UNLIMITED</unit>
        </duration>
        <recurring>
            <billingPeriod>MONTHLY</billingPeriod>
            <recurringPrice>
                <price>
                    <currency>USD</currency>
                    <value>5.00</value>
                </price>
            </recurringPrice>
        </recurring>
        <usages>
            <usage name="smallQ-monthly-usage" billingMode="IN_ARREAR"
usageType="CONSUMABLE">
                <billingPeriod>MONTHLY</billingPeriod>
                <tiers>
                    <tier>
                        <blocks>
                            <tieredBlock>
                                <unit>words-count</unit>
                                <size>100000</size>
                                <prices>
                                    <price>
                                        <currency>USD</currency>
                                        <value>0</value>
                                    </price>
                                </prices>
                                <max>100000</max>
                            </tieredBlock>
                        </blocks>
                    </tier>
                    <tier>
                        <blocks>
                            <tieredBlock>
                                <unit>words-count</unit>
                                <size>10000</size>
                                <prices>
                                    <price>
                                        <currency>USD</currency>
                                        <value>1.00</value>
                                    </price>
                                </prices>
                                <max>1000000000</max>
                            </tieredBlock>
                        </blocks>
                    </tier>
                </tiers>
            </usage>
        </usages>
    </finalPhase>
</plan>
<plan name="smallQ-monthly-without-trial">
    <product>smallQ2</product>
    <finalPhase type="EVERGREEN">
        <duration>
            <unit>UNLIMITED</unit>
        </duration>
        <recurring>
            <billingPeriod>MONTHLY</billingPeriod>
            <recurringPrice>
                <price>
                    <currency>USD</currency>
                    <value>5.00</value>
                </price>
            </recurringPrice>
        </recurring>
        <usages>
            <usage name="smallQ-monthly-without-trial-usage"
billingMode="IN_ARREAR" usageType="CONSUMABLE">
                <billingPeriod>MONTHLY</billingPeriod>
                <tiers>
                    <tier>
                        <blocks>
                            <tieredBlock>
                                <unit>words-count</unit>
                                <size>100000</size>
                                <prices>
                                    <price>
                                        <currency>USD</currency>
                                        <value>0</value>
                                    </price>
                                </prices>
                                <max>100000</max>
                            </tieredBlock>
                        </blocks>
                    </tier>
                    <tier>
                        <blocks>
                            <tieredBlock>
                                <unit>words-count</unit>
                                <size>10000</size>
                                <prices>
                                    <price>
                                        <currency>USD</currency>
                                        <value>1.00</value>
                                    </price>
                                </prices>
                                <max>1000000000</max>
                            </tieredBlock>
                        </blocks>
                    </tier>
                </tiers>
            </usage>
        </usages>
    </finalPhase>
</plan>
<plan name="smallQ-monthly-without-usage">
    <product>smallQ3</product>
    <finalPhase type="EVERGREEN">
        <duration>
            <unit>UNLIMITED</unit>
        </duration>
        <recurring>
            <billingPeriod>MONTHLY</billingPeriod>
            <recurringPrice>
                <price>
                    <currency>USD</currency>
                    <value>5.00</value>
                </price>
            </recurringPrice>
        </recurring>
    </finalPhase>
</plan>





On Mon, Feb 15, 2016 at 11:27 PM, Stéphane Brossier <
notifications@github.com> wrote:

> @husam-otri <https://github.com/husam-otri> I see that your Plan has a
> TRIAL. Is the charged_through_date null when you are in TRIAL phase or is
> it always null even as you move to EVERGREEN phase? Does it have anything
> to do with usage (or if you remove usage section) you get same behavior?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/killbill/killbill/issues/490#issuecomment-184402795>.
>



-- 
Best Regards,
Husam

**Comment by sbrossie** [comment]
Confused... I created a very simple catalog based on your input:

```

<catalog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="CatalogSchema.xsd ">

  <effectiveDate>2016-01-01T00:00:00+00:00</effectiveDate>
  <catalogName>Catalog490</catalogName>

  <recurringBillingMode>IN_ADVANCE</recurringBillingMode>

  <currencies>
    <currency>USD</currency>
  </currencies>


  <products>
    <product name="smallQ3">
      <category>BASE</category>
    </product>
  </products>

  <rules>
    <changePolicy>
      <changePolicyCase>
        <policy>END_OF_TERM</policy>
      </changePolicyCase>
    </changePolicy>
    <changeAlignment>
      <changeAlignmentCase>
        <alignment>START_OF_SUBSCRIPTION</alignment>
      </changeAlignmentCase>
    </changeAlignment>
    <cancelPolicy>
      <cancelPolicyCase>
        <productCategory>BASE</productCategory>
        <policy>END_OF_TERM</policy>
      </cancelPolicyCase>
    </cancelPolicy>
    <createAlignment>
      <createAlignmentCase>
        <alignment>START_OF_BUNDLE</alignment>
      </createAlignmentCase>
    </createAlignment>
    <billingAlignment>
      <billingAlignmentCase>
        <alignment>ACCOUNT</alignment>
      </billingAlignmentCase>
    </billingAlignment>
    <priceList>
      <priceListCase>
        <toPriceList>DEFAULT</toPriceList>
      </priceListCase>
    </priceList>
  </rules>
  <plans>
    <plan name="smallQ-monthly-without-usage">
      <product>smallQ3</product>
      <finalPhase type="EVERGREEN">
        <duration>
          <unit>UNLIMITED</unit>
        </duration>
        <recurring>
          <billingPeriod>MONTHLY</billingPeriod>
          <recurringPrice>
            <price>
              <currency>USD</currency>
              <value>5.00</value>
            </price>
          </recurringPrice>
        </recurring>
      </finalPhase>
    </plan>
  </plans>
  <priceLists>
    <defaultPriceList name="DEFAULT">
      <plans>
        <plan>smallQ-monthly-without-usage</plan>
      </plans>
    </defaultPriceList>
  </priceLists>
</catalog>

```

And then i created a small test in our [integration test framework](https://github.com/killbill/killbill-integration-tests/tree/master/killbill-integration-tests):

```

# encoding: utf-8

$LOAD_PATH.unshift File.expand_path('../..', __FILE__)

require 'test_base'

module KillBillIntegrationTests

  class Issue490 < Base


    def setup

      setup_base(self.method_name, DEFAULT_MULTI_TENANT_INFO, "2016-02-01T08:00:00.000Z", '127.0.0.1', '8080')

      upload_catalog('Catalog-490.xml', false, @user, @options)
      @account = create_account(@user, @options)
    end

    def teardown
      teardown_base
    end

    def test_charged_through_date

      bp = create_entitlement_base(@account.account_id, 'smallQ3', 'MONTHLY', 'DEFAULT',  @user, @options)
      check_entitlement(bp, 'smallQ3', 'BASE', 'MONTHLY', 'DEFAULT', '2016-02-01', nil)
      wait_for_expected_clause(1, @account, @options, &@proc_account_invoices_nb)
      assert_equal('2016-03-01', bp.charged_through_date)
    end

  end
end

```

And that shows that the `charged_through_date` is **non null** and contains the expected value **`2016-02-01`**.

**Comment by husam-otri** [comment]
Confused also, so 
- I used the same catalog you pasted above (killbill 0.16.x) , but _charged_through_date_ still null   
- then i downgrade to killbill version 0.14.0 then _charged_through_date_ _2016-03-16 07:25:09_ populated!

**Comment by husam-otri** [comment]
Hey @sbrossie There is another bug - in v0.16.1 - and maybe not related to this ticket, so after i do the subscription i can see new record in _subscriptions_ table But i can't see any invoice in _invoices and also there is no any payment in _payments table_  
Killbill version **0.16.1**
**Catalog**: the same simple catalog which you shared in your last comment above _Catalog490_

**Comment by sbrossie** [comment]
@husam-otri Could you verify that your system is correctly installed. In particular if you move from 0.14.x to 0.16.x you need to [upgrade the system](https://github.com/killbill/killbill/wiki/Kill-Bill-Migration-0.14.x-to-0.16.x).

Are you saying any error, exceptions as you are running those curls?

**Comment by husam-otri** [comment]
Yeah, I run all quires in order to upgrade from 0.14.0 to 0.16.1
<https://github.com/killbill/killbill/wiki/Kill-Bill-Migration-0.14.x-to-0.16.x>
block
by block and i didn't see any error, also to make sure to make more sure i
dropped the old DB then create fresh DB using
http://docs.killbill.io/0.16/ddl.sql unfortunately i have got the same
result!

On Tue, Feb 16, 2016 at 6:59 PM, Stéphane Brossier <notifications@github.com
> wrote:

> @husam-otri <https://github.com/husam-otri> Could you verify that your
> system is correctly installed. In particular if you move from 0.14.x to
> 0.16.x you need to upgrade the system
> <https://github.com/killbill/killbill/wiki/Kill-Bill-Migration-0.14.x-to-0.16.x>
> .
>
> Are you saying any error, exceptions as you are running those curls?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/killbill/killbill/issues/490#issuecomment-184770212>.
>



-- 
Best Regards,
Eng Husam Al-Otri
Software Engineering

**Comment by sbrossie** [comment]
@husam-otri I m planning on (re) trying with 0.16.1 to see if this is specific with that release. I'll let you know. Any news on your side?

**Comment by sbrossie** [comment]
@husam-otri I just setup a Kill Bill instance with version `0.16.1`, and retried the test above, and things work fine. Most probably this is an issue with your setup/upgrade.

I am not sure how to help here. A few suggestions:
1. Try to run the test i provided (i can provide instructions if you care)
2. Debug the code (i can also provide instructions on where to place breakpoints if you care)

**Comment by husam-otri** [comment]
1 +2 ) yes please if you provide steps how to run the test and how to debug the code that would be really helpful ... thanks in advance

however i notice this issue in the log, (please note i am using also stripe plugin **3.0.2** and exactly the same catalog which you shared it above )

2016-02-20 16:32:15,297 [http-bio-8080-exec-8] INFO  c.s.j.a.c.filter.LoggingFilter - 47 * Server in-bound request
47 > POST http://127.0.0.1:8080/1.0/kb/subscriptions
47 > host: 127.0.0.1:8080
47 > x-killbill-apikey: bob
47 > accept: application/json
47 > x-killbill-apisecret: lazar
47 > x-killbill-createdby: 360bddb0-25eb-4dd6-894b-f5b91cbdc58b
47 > content-type: application/json
47 > authorization: Basic YWRtaW46cGFzc3dvcmQ=
47 > connection: keep-alive
47 > user-agent: NING/1.0
47 > content-length: 154
47 > 

2016-02-20 16:32:15,670 [http-bio-8080-exec-8] INFO  c.s.j.a.c.filter.LoggingFilter - 47 * Server out-bound response
47 < 201
47 < Location: http://127.0.0.1:8080/1.0/kb/subscriptions/cf14180a-3714-4251-8b6e-516dd192f5e7
47 < Content-Type: application/json
47 < 

2016-02-20 16:32:15,713 [bus_events-th] INFO  o.k.b.j.p.b.BillCycleDayCalculator - Calculated BCD: subscription id cf14180a-3714-4251-8b6e-516dd192f5e7, subscription start 2016-03-21T14:32:15.000Z, timezone UTC, bcd UTC 21, bcd local 21
2016-02-20 16:32:15,838 [bus_events-th] ERROR o.k.b.j.p.b.DefaultInternalBillingApi - Failing to identify catalog components while creating BillingEvent from transition: 2aa7763a-d85d-47b4-b179-1c4203ceb6de
org.killbill.billing.catalog.api.CatalogApiException: This price does not have a value for the currency 'null'.
	at org.killbill.billing.catalog.DefaultInternationalPrice.getPrice(DefaultInternationalPrice.java:78)
	at org.killbill.billing.junction.plumbing.billing.DefaultBillingEvent.getFixedPrice(DefaultBillingEvent.java:317)
	at org.killbill.billing.junction.plumbing.billing.DefaultBillingEvent.<init>(DefaultBillingEvent.java:79)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.addBillingEventsForSubscription(DefaultInternalBillingApi.java:208)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.addBillingEventsForBundles(DefaultInternalBillingApi.java:162)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.getBillingEventsForAccountAndUpdateAccountBCD(DefaultInternalBillingApi.java:108)
	at org.killbill.billing.invoice.InvoiceDispatcher.processAccountWithLock(InvoiceDispatcher.java:235)
	at org.killbill.billing.invoice.InvoiceDispatcher.processAccount(InvoiceDispatcher.java:213)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionInternal(InvoiceDispatcher.java:199)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionForInvoiceGeneration(InvoiceDispatcher.java:174)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionForInvoiceGeneration(InvoiceDispatcher.java:170)
	at org.killbill.billing.invoice.InvoiceListener.handleSubscriptionTransition(InvoiceListener.java:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.EventBusThatThrowsException.dispatchWithException(EventBusThatThrowsException.java:106)
	at com.google.common.eventbus.EventBusThatThrowsException.dispatchQueuedEventsWithException(EventBusThatThrowsException.java:94)
	at com.google.common.eventbus.EventBusThatThrowsException.postWithException(EventBusThatThrowsException.java:82)
	at org.killbill.bus.DefaultPersistentBus.dispatchBusEventWithMetrics(DefaultPersistentBus.java:299)
	at org.killbill.bus.dispatching.BusCallableCallback.dispatch(BusCallableCallback.java:40)
	at org.killbill.bus.dispatching.BusCallableCallback.dispatch(BusCallableCallback.java:29)
	at org.killbill.queue.dispatching.Dispatcher$CallableQueue.call(Dispatcher.java:102)
	at org.killbill.queue.dispatching.Dispatcher$CallableQueue.call(Dispatcher.java:85)
	at org.killbill.commons.concurrent.WrappedCallable.call(WrappedCallable.java:42)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.killbill.commons.concurrent.WrappedRunnable.run(WrappedRunnable.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2016-02-20 16:32:15,839 [bus_events-th] INFO  o.k.b.j.p.b.BillCycleDayCalculator - Calculated BCD: subscription id cf14180a-3714-4251-8b6e-516dd192f5e7, subscription start 2016-03-21T14:32:15.000Z, timezone UTC, bcd UTC 21, bcd local 21
2016-02-20 16:32:15,851 [bus_events-th] ERROR o.k.b.j.p.b.DefaultInternalBillingApi - Failing to identify catalog components while creating BillingEvent from transition: 01836945-509d-4244-beed-c4b24a883695
org.killbill.billing.catalog.api.CatalogApiException: This price does not have a value for the currency 'null'.
	at org.killbill.billing.catalog.DefaultInternationalPrice.getPrice(DefaultInternationalPrice.java:78)
	at org.killbill.billing.junction.plumbing.billing.DefaultBillingEvent.getRecurringPrice(DefaultBillingEvent.java:321)
	at org.killbill.billing.junction.plumbing.billing.DefaultBillingEvent.<init>(DefaultBillingEvent.java:80)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.addBillingEventsForSubscription(DefaultInternalBillingApi.java:208)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.addBillingEventsForBundles(DefaultInternalBillingApi.java:162)
	at org.killbill.billing.junction.plumbing.billing.DefaultInternalBillingApi.getBillingEventsForAccountAndUpdateAccountBCD(DefaultInternalBillingApi.java:108)
	at org.killbill.billing.invoice.InvoiceDispatcher.processAccountWithLock(InvoiceDispatcher.java:235)
	at org.killbill.billing.invoice.InvoiceDispatcher.processAccount(InvoiceDispatcher.java:213)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionInternal(InvoiceDispatcher.java:199)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionForInvoiceGeneration(InvoiceDispatcher.java:174)
	at org.killbill.billing.invoice.InvoiceDispatcher.processSubscriptionForInvoiceGeneration(InvoiceDispatcher.java:170)
	at org.killbill.billing.invoice.InvoiceListener.handleSubscriptionTransition(InvoiceListener.java:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
	at com.google.common.eventbus.EventBusThatThrowsException.dispatchWithException(EventBusThatThrowsException.java:106)
	at com.google.common.eventbus.EventBusThatThrowsException.dispatchQueuedEventsWithException(EventBusThatThrowsException.java:94)
	at com.google.common.eventbus.EventBusThatThrowsException.postWithException(EventBusThatThrowsException.java:82)
	at org.killbill.bus.DefaultPersistentBus.dispatchBusEventWithMetrics(DefaultPersistentBus.java:299)
	at org.killbill.bus.dispatching.BusCallableCallback.dispatch(BusCallableCallback.java:40)
	at org.killbill.bus.dispatching.BusCallableCallback.dispatch(BusCallableCallback.java:29)
	at org.killbill.queue.dispatching.Dispatcher$CallableQueue.call(Dispatcher.java:102)
	at org.killbill.queue.dispatching.Dispatcher$CallableQueue.call(Dispatcher.java:85)
	at org.killbill.commons.concurrent.WrappedCallable.call(WrappedCallable.java:42)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.killbill.commons.concurrent.WrappedRunnable.run(WrappedRunnable.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2016-02-20 16:32:15,856 [bus_events-th] INFO  o.k.b.j.p.b.DefaultInternalBillingApi - Computed billing events for accountId 360bddb0-25eb-4dd6-894b-f5b91cbdc58b

**Comment by sbrossie** [comment]
@husam-otri Did you specify a currency when you created your account? We released the constraints in 0.16.x to not enforce 'currency` for accounts (which mean by default they are null). It also mean that when the system attempts to generate an invoice for which account has no currency, it will fail (with the error you reported above).

Then, from there, failure to generate the invoice will also trigger a failure in setting the `chargedThroughDate`.

**Comment by husam-otri** [comment]
Yeah, its working now after i provide "currency" in the payload, thanks @sbrossie , but i can't find where you mentioned in documentation this really important information "_We released the constraints in 0.16.x to not enforce 'currency` for accounts_"  ?

because its really not easy to know that failure generating invoice related to add "currency" field when create new account!

2) assume we would support more that 1 currency, how we can provide list of currencies to that create account API, it takes only String Not list of String?

**Comment by sbrossie** [comment]
@husam-otri answers to your questions:

> Yeah, its working now after i provide "currency" in the payload, thanks @sbrossie , but i can't find where you mentioned in documentation this important information "We released the constraints in 0.16.x to not enforce 'currency` for accounts" ?

The information about releasing the constraint exists in the [migration page](https://github.com/killbill/killbill/wiki/Kill-Bill-Migration-0.14.x-to-0.16.x):

```
alter table accounts modify currency varchar(3) default null;
```

However you are absolutely correct that it requires paying a lot of attention when applying the changes, and nowhere is it written in black and white that "We released the constraints in 0.16.x to not enforce 'currency` for accounts" 

Between even revisions there are usually quite a few (incompatible) changes. We do our best (with the minimal set of resources we have) to document them and make the migration easy. We are fully aware that what we currently provide is the bare minimum and as we keep moving forward (and as we get more and more folks using the system), the releases notes and potentially migration scripts will become a priority.


Now coming to your `currency` issue, i am still puzzled why that worked for 0.14.x., or does it mean you changed the way you created your accounts? In `0.14.x`, you provided the `currency` which was mandatory and in  `0.16.x` you stopped passing the `currency` (just because this is *not* mandatory)? Why?



> 2) assume we would support more that 1 currency, how we can provide list of currencies to that create account API, it takes only String Not list of String?

Not sure i understand your use case. How would the system know which one to use? Also, could you use the mailing list for questions (this goes beyond solving the issue you mentioned in that ticket).
